### ISSUE TITLE
Prospector bug

### ISSUE THREAD

**Issue by weepingwillowben** [PettingZoo-Team/PettingZoo#122]
Getting this error message in prospector very, very rarely. 
```
From cffi callback <function CollisionHandler._set_begin.<locals>.cf at 0x7fb3ddf3f598>:
Traceback (most recent call last):
  File "/home/ben/.virtualenvs/zoo/lib/python3.6/site-packages/pymunk/collision_handler.py", line 64, in cf
    x = func(Arbiter(_arb, self._space), self._space, self._data)
  File "/home/ben/class_projs/PettingZoo/pettingzoo/butterfly/prospector/prospector.py", line 459, in handoff_gold_handler
    # This collision handler is only for prospector -> banker gold handoffs
AttributeError: 'NoneType' object has no attribute 'parent_body'
```

It doesn't seem to cause a crash, and could probably be ignored, but it doesn't seem good just to leave it generating these errors.

**Comment by justinkterry** [comment]
@yashaslokesh

**Comment by justinkterry** [comment]
Fixed in #125
